j.n.SocketHelpers#ipToHost no longer requests info it does not use.

LeeTibbert opened the pull request:
We make a small, focused change to specify the 'service' argument
in a way that instructs getnameinfo() to not spend time/cycles retrieving
that information. 

Code is restructured to unify almost identical calls.

For consistency, a hard-coded number is converted to the apparently intended constant.

See Issue #2311 for a discussion of further improvements which could be made.

ekrich reviewed (changes requested):
I think this looks good. Just out of curiosity, can you just put `null` for service at the call site and avoid the comments completely? Otherwise, rather than duplicate the comment, maybe put it above the definition of `service`.

LeeTibbert commented:
@ekrich Thank you for the timely review. Changed as suggested. It does make the code simpler. 

I removed the sole comment referencing 'service'. I did a search on the file for other occurrences
of the word 'service' and could find none.  Did you find an actual duplicate comment in the code
you reviewed or were you implying one (above the 'service' declaration or somewhere else?)
My command of English is failing me these days.

ekrich reviewed (changes requested):
I guess I looked a little closer after the first review.

ekrich commented on posixlib/src/main/scala/scala/scalanative/runtime/SocketHelpers.scala:
```diff
@@ -183,38 +183,56 @@ object SocketHelpers {
+      }
+
+    // See error reporting note at bottom of sole caller ipToHost().
+    inet_pton(addr.sa_family.toInt, src, dst) == 1
```
Whatever this does could be a clue to the name of the function that calls it. Is this function returning 1 like true in C so that is good?

LeeTibbert replied:
Reworked to be less C and more Scala idiomatic.

LeeTibbert replied:
Re-worked again.  Gave helper method `tailorSockaddr()` a name which, I believe, 
better describes its intended use. 

I tried making that method local rather than private but the results left
the body of `ipToHost()` as nearly as long and confusing as the code this PR
wants to replace.

ekrich replied:
@LeeTibbert This satisfies my ideas - I like the Boolean and the naming better now.

ekrich reviewed (approved):
LTGM, maybe just a review to make sure @WojciechMazur is happy with it. To me it is a definite improvement.
